terraform/aws/best-practice/aws-s3-bucket-versioning-not-enabled.tf

```diff
@@ -0,0 +1,108 @@
+# ruleid: aws-s3-bucket-versioning-not-enabled
+resource "aws_s3_bucket" "fail4" {
+  region        = "us-west-2"
+  bucket        = "my_bucket"
+  acl           = "public-read"
+  force_destroy = true
+  tags {
+    Name = "my-bucket"
+  }
+}
+# ruleid: aws-s3-bucket-versioning-not-enabled
+resource "aws_s3_bucket" "fail3" {
+  region        = "us-west-2"
+  bucket        = "my_bucket"
+  acl           = "public-read"
+  force_destroy = true
+  tags          = { Name = "my-bucket" }
+  enabled       = True
+}
+# ruleid: aws-s3-bucket-versioning-not-enabled
+resource "aws_s3_bucket" "fail2" {
+  region        = "us-west-2"
+  bucket        = "my_bucket"
+  acl           = "public-read"
+  force_destroy = true
+  tags = {
+    Name = "my-bucket"
+    wrong_field = {
+    enabled = true }
+  }
+}
+# ruleid: aws-s3-bucket-versioning-not-enabled
+resource "aws_s3_bucket" "fail" {
+  region        = "us-west-2"
+  bucket        = "my_bucket"
+  acl           = "public-read"
+  force_destroy = true
+  tags          = { Name = "my-bucket" }
+  wrong_field   = { versioning = { enabled = true } }
+}
+
+
+resource "aws_s3_bucket" "pass" {
+  region        = "us-west-2"
+  bucket        = "my_bucket"
+  acl           = "public-read"
+  force_destroy = true
+
+  tags = { Name = "my-bucket" }
+
+  logging {
+    target_bucket = "logging-bucket"
+    target_prefix = "log/"
+  }
+  versioning {
+    enabled = true
+  }
+}
+# ruleid: aws-s3-bucket-versioning-not-enabled
+resource "aws_s3_bucket" "failcomplex" {
+  acl    = "public-read-write"
+  bucket = "superfail"
+
+  versioning {
+    enabled    = false
+    mfa_delete = false
+  }
+
+  policy = <<POLICY
+{
+  "Version":"2012-10-17",
+  "Statement":[
+    {
+      "Sid":"AddCannedAcl",
+      "Effect":"Allow",
+      "Principal": "*",
+      "Action":["s3:PutObject","s3:PutObjectAcl"],
+      "Resource":"arn:aws:s3:::superfail/*"
+    },
+    {
+        "Principal": {
+            "AWS": ["*"],
+            "Effect": "Deny",
+            "Action": [
+                "s3:*"
+            ],
+            "Resource": [
+                "*"
+            ]
+        }
+    }
+  ]
+}
+POLICY
+}
+
+
+resource "aws_s3_bucket" "this" {
```

Comment:
This one also gets flagged. Could try filtering it out with `enabled = var.$ENABLED`.

Sadly we don't have TF variable resolution yet... but maybe someday!